parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToType.java

```diff
@@ -200,7 +200,7 @@ protected int nextId() {
     return current;
   }
 
-  private int getId(org.apache.parquet.schema.Type type) {
+  protected int getId(org.apache.parquet.schema.Type type) {
```

Comment:
Why was this changed?

Reply:
This is changed so that we can override `getid` and use that to detect whether a schema has ID or not.

Reply:
I think it might be better to make a new visitor for this, rather than reusing `MessageTypeToType`. That maintains independence between classes that don't have the same purpose. It was easy before to use `MessageTypeToType` without modification, but there's no need to do that if it is no longer easy.

Reply:
OK, will add new one.